external/storm-mqtt/core/src/test/java/org/apache/storm/mqtt/StormMqttIntegrationTest.java

```diff
@@ -0,0 +1,132 @@
+import org.slf4j.LoggerFactory;
+
+import java.io.Serializable;
+import java.net.URI;
+import java.util.Arrays;
+
+@Category(IntegrationTest.class)
+public class StormMqttIntegrationTest implements Serializable{
+    private static final Logger LOG = LoggerFactory.getLogger(StormMqttIntegrationTest.class);
+    private static BrokerService broker;
+
+
+    @AfterClass
+    public static void cleanup() throws Exception {
+        broker.stop();
+    }
+
+    @BeforeClass
+    public static void start() throws Exception {
+        LOG.warn("Starting broker...");
+        broker = new BrokerService();
+        broker.addConnector("mqtt://localhost:1883");
+        broker.setDataDirectory("target");
+        broker.start();
+        LOG.debug("MQTT broker started");
+    }
+
+
+    @Test
+    public void testMqttTopology() throws Exception {
+        MQTT client = new MQTT();
+        client.setTracer(new MqttLogger());
+        URI uri = URI.create("tcp://localhost:1883");
+        client.setHost(uri);
+
+        client.setClientId("MQTTSubscriber");
+        client.setCleanSession(false);
+        BlockingConnection connection = client.blockingConnection();
+        connection.connect();
+        Topic[] topics = {new Topic("/integration-result", QoS.AT_LEAST_ONCE)};
+        byte[] qoses = connection.subscribe(topics);
+
+        LocalCluster cluster = new LocalCluster();
+        cluster.submitTopology("test", new Config(), buildMqttTopology());
+
+        System.out.println("topology started");
+        Thread.sleep(10000);
+
+        // publish a retained message to the broker
+        MqttOptions options = new MqttOptions();
+        options.setCleanConnection(false);
+        MqttPublisher publisher = new MqttPublisher(options);
+        publisher.connectMqtt("MqttPublisher");
+        publisher.publish(new MqttMessage("/mqtt-topology", "test".getBytes()), true);
+
+        LOG.info("published message");
+
+        Message message = connection.receive();
+        LOG.info("Message recieved on topic: " + message.getTopic());
+        LOG.info("Payload: " + new String(message.getPayload()));
+        message.ack();
+
+        Assert.assertArrayEquals(message.getPayload(), "hello mqtt".getBytes());
+        Assert.assertEquals(message.getTopic(), "/integration-result");
+        cluster.shutdown();
+    }
+
+    public StormTopology buildMqttTopology(){
+        TopologyBuilder builder = new TopologyBuilder();
+
+        MqttOptions options = new MqttOptions();
+        options.setTopics(Arrays.asList("/mqtt-topology"));
+        options.setCleanConnection(false);
+        MqttSpout spout = new MqttSpout(new StringMessageMapper(), options);
+
+        MqttBolt bolt = new MqttBolt(options, new MqttTupleMapper() {
+            @Override
+            public MqttMessage toMessage(ITuple tuple) {
+                LOG.info("Received: " + tuple);
+                return new MqttMessage("/integration-result", "hello mqtt".getBytes());
```

Comment:
You may want to extract constants for topic names like "/integration-result", "/mqtt-topology"  as they are used at multiple places.